src/auth/__tests__/index.spec.js

```diff
@@ -133,35 +138,169 @@ describe('auth', () => {
-        unexpectedError
+        unexpectedError,
       );
       expect.assertions(1);
       const parameters = {
         code: 'a code',
         verifier: 'a verifier',
         redirectUri,
         state,
-        scope: 'openid'
+        scope: 'openid',
       };
       await expect(auth.exchange(parameters)).rejects.toMatchSnapshot();
     });
   });
 
+  describe('passwordless flow', () => {
+    describe('with email connection', () => {
+      it('should begin with code', async () => {
+        fetchMock.postOnce(
+          'https://samples.auth0.com/passwordless/start',
+          emptySuccess,
+        );
+        expect.assertions(1);
+        await auth.passwordlessWithEmail({
+          email: 'info@auth0.com',
+          send: 'link',
+        });
+        expect(fetchMock.lastCall()).toMatchSnapshot();
+      });
+
+      it('should begin with link', async () => {
+        fetchMock.postOnce(
+          'https://samples.auth0.com/passwordless/start',
+          emptySuccess,
+        );
+        expect.assertions(1);
+        await auth.passwordlessWithEmail({
+          email: 'info@auth0.com',
+          send: 'link',
+        });
+        expect(fetchMock.lastCall()).toMatchSnapshot();
+      });
+
+      it('should begin with optional parameters', async () => {
+        fetchMock.postOnce(
+          'https://samples.auth0.com/passwordless/start',
+          emptySuccess,
+        );
+        expect.assertions(1);
+        await auth.passwordlessWithEmail({
+          email: 'info@auth0.com',
+          send: 'code',
+          authParams: {
+            scope: 'openid profile',
+          },
+        });
+        expect(fetchMock.lastCall()).toMatchSnapshot();
+      });
+
+      it('should continue', async () => {
+        fetchMock.postOnce('https://samples.auth0.com/oauth/token', tokens);
+        expect.assertions(1);
+        await auth.loginWithEmail({
+          email: 'info@auth0.com',
+          code: '123456',
+        });
+        expect(fetchMock.lastCall()).toMatchSnapshot();
+      });
+
+      it('should continue with optional parameters', async () => {
+        fetchMock.postOnce('https://samples.auth0.com/oauth/token', tokens);
+        expect.assertions(1);
+        await auth.loginWithEmail({
+          email: 'info@auth0.com',
+          code: '123456',
+          audience: 'http://myapi.com',
+          scope: 'openid',
+        });
+        expect(fetchMock.lastCall()).toMatchSnapshot();
+      });
+    });
+
+    describe('with SMS connection', () => {
+      it('should begin with code', async () => {
+        fetchMock.postOnce(
+          'https://samples.auth0.com/passwordless/start',
+          emptySuccess,
+        );
+        expect.assertions(1);
+        await auth.passwordlessWithSMS({
+          phoneNumber: '+5491159991000',
+          send: 'code',
+        });
+        expect(fetchMock.lastCall()).toMatchSnapshot();
+      });
+
+      it('should begin with link', async () => {
```

Comment:
If we're removing the link from SMS, we should remove this test as well.

Reply:
We're not removing support for `link`. We're just not documenting it 👍 Tests should still prove the endpoint can be called with additional auth params and/or with link

Reply:
But tests are documentation, and it may be confusing for a customer to come across this.